### PR TITLE
scas: init at 0.4.6

### PR DESCRIPTION
**Pull request by siraben**
###### Motivation for this change
Add scas Z80 assembler

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/compilers/scas/default.nix [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  src = fetchFromGitHub {
+    owner = "KnightOS";
+    repo = "scas";
+    rev = "${version}";
+    sha256 = "1c6s9nivbwgv0f8n7j73h54ydgqw5dcpq8l752dfrnqg3kv3nn0h";
+  };
+  nativeBuildInputs = [ cmake ];
+  installTargets = [ "install" ];
+  hardeningDisable = [ "format" ];
+    meta = with stdenv.lib; {
```
Comment (syberant) [inline pkgs/development/compilers/scas/default.nix:16]:
```suggestion
  meta = with stdenv.lib; {
```
Tiny nitpick here, meta has the wrong indentation level.

File: pkgs/development/compilers/scas/default.nix [hunk 2]
```diff
@@ -0,0 +1,22 @@
+{ fetchFromGitHub, stdenv, cmake }:
+
+
+stdenv.mkDerivation rec {
+  pname = "scas";
+  version = "0.4.6";
+  src = fetchFromGitHub {
+    owner = "KnightOS";
+    repo = "scas";
+    rev = "${version}";
```
Comment (sikmir) [inline pkgs/development/compilers/scas/default.nix:10]:
```suggestion
    rev = version;
```

File: pkgs/development/compilers/scas/default.nix [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    sha256 = "1c6s9nivbwgv0f8n7j73h54ydgqw5dcpq8l752dfrnqg3kv3nn0h";
+  };
+  nativeBuildInputs = [ cmake ];
+  installTargets = [ "install" ];
+  hardeningDisable = [ "format" ];
+    meta = with stdenv.lib; {
+    homepage    = "https://knightos.org/";
+    description = "Assembler and linker for the Z80.";
+    license     = licenses.free;
```
Comment (sikmir) [inline pkgs/development/compilers/scas/default.nix:19]:
```suggestion
    license     = licenses.mit;
```

File: pkgs/development/compilers/scas/default.nix [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+  pname = "scas";
+  version = "0.4.6";
+  src = fetchFromGitHub {
+    owner = "KnightOS";
+    repo = "scas";
+    rev = "${version}";
+    sha256 = "1c6s9nivbwgv0f8n7j73h54ydgqw5dcpq8l752dfrnqg3kv3nn0h";
+  };
+  nativeBuildInputs = [ cmake ];
+  installTargets = [ "install" ];
```
Comment (syberant) [inline pkgs/development/compilers/scas/default.nix:14]: Is setting `installTargets` necessary? If I'm not mistaken the build script automatically uses the install target.

Comment (sikmir) [inline pkgs/development/compilers/scas/default.nix:14]: Is it really needed? "install" is default target.

**Review by sikmir — APPROVED** [review]

**Comment by siraben** [comment]
Ready to merge.

**Comment by Infinisil** [comment]
Please rebase the PR such that it has an initial commit that adds you as a maintainer and a second commit that adds the package

**Pull request merged**